sys/random/random.c

```diff
@@ -54,3 +54,36 @@ void random_bytes(uint8_t *target, size_t n)
         *target++ = *random_pos++;
     }
 }
+
+uint32_t random_uint32_range(uint32_t a, uint32_t b)
+{
+    uint32_t range = b - a;
+    uint32_t highest_pow2 = (1 << bitarithm_msb(range));
+    uint32_t random = random_uint32();
+
+    /* check if range is a power of two */
+    if (range && !(range & (range - 1))) {
```

Comment:
> [edit2]: but if range is pow2 wouldn't that mean range == highest_pow2 then? If so it might be even more efficient to compare those instead of the bit operation?!

Tried that, but results are same (well, "my" solution produced even two more numbers in 20 seconds, but I guess this is below any optimization influence anyway)

Reply:
yeah I think nowadays its hard to outsmart the compiler 😉 So best we can do is go for well documented and structured code.